Algo.h

```diff
@@ -294,7 +385,8 @@ class MMP
                 // invert wale pains???
 
                 for (auto &e : ((Vertex *)(source->p))->edges) {
-                    Interval ii(0, 0, 0, e->length(), 0, NULL, e);
+                    //check for invert
+                    Interval ii(0, 0, 0, e->length(), 0, NULL, e, false);
```

Comment:
In the function initialize(), aren't we supposed to add the edges **opposite** to the current vertex, and not the adjacent ones?